src/Lumi/Components/Form.purs

```diff
@@ -428,15 +428,20 @@ checkbox props =
                   { style: R.css { alignItems: "center" }
                   , children:
                       [ if readonly
-                          then Input.input Input.checkbox
-                                { style = R.css { marginBottom: "0" }
-                                , checked = Input.Off
-                                }
-                          else Input.input Input.checkbox
-                                 { style = R.css { marginBottom: "0" }
-                                 , checked = if value then Input.On else Input.Off
-                                 , onChange = Events.handler (stopPropagation >>> targetChecked) (traverse_ onChange)
-                                 }
+                          then
+                            Input.input Input.checkbox
+                              { style = R.css { marginBottom: "0" }
+                              , checked = if value then Input.On else Input.Off
+                              , disabled = true
+                              , onChange = Events.handler Events.syntheticEvent \_ -> pure unit
+                              }
+                          else
+                            Input.input Input.checkbox
+                              { style = R.css { marginBottom: "0" }
+                              , checked = if value then Input.On else Input.Off
+                              , disabled = false
```

Comment:
I think if it's disabled it can't be changed, so the surrounding if/else could probably just be on this line rather than defining the input twice